app/src/ui/diff/code-mirror-host.tsx

```diff
@@ -164,6 +164,12 @@ export class CodeMirrorHost extends React.Component<ICodeMirrorHostProps, {}> {
 
     CodeMirrorHost.updateDoc(this.codeMirror, this.props.value)
     this.resizeObserver.observe(this.codeMirror.getWrapperElement())
+
+    document.addEventListener('dialog-appeared', this.onDialogAppeared)
```

Comment:
For context, Codemirror was rendering before the dialog was fully mounted and in the scenario of having no scroll bars on either the file list and the diff, the diff would not render well the first time. This event listener is to make code mirror rerender after dialog has appeared and size figured out. You may see a quick flicker when opening in unified mode. This is a quick solution to make code mirror rerender. When we switch to the split diff engine, this won't be needed anymore.

Reply:
> When we switch to the split diff engine, this won't be needed anymore.

Oh lol I guess I spent too long writing my review and missed this 😂

Reply:
Yeah. But, you still have a valid point - I am out of time for now, but will see if I can do like you said and narrow it to code mirror instances inside a dialog.